lib/fog/vsphere/requests/compute/vm_clone.rb

```diff
@@ -581,31 +587,57 @@ def vm_clone(options = {})
+            srm = @connection.serviceContent.storageResourceManager
+            result = srm.RecommendDatastores(:storageSpec => storage_spec)
+
+            # if result array contains recommendation, we can apply it
+            if key = result.recommendations.first.key
+              task = srm.ApplyStorageDrsRecommendation_Task(:key => [key])
+              result = task.wait_for_completion
+              new_vm = result.vm
+            end
+          else
+            task = vm_mob_ref.CloneVM_Task(:folder => dest_folder,
                                          :name => options['name'],
                                          :spec => clone_spec)
-          # Waiting for the VM to complete allows us to get the VirtulMachine
-          # object of the new machine when it's done.  It is HIGHLY recommended
-          # to set 'wait' => true if your app wants to wait.  Otherwise, you're
-          # going to have to reload the server model over and over which
-          # generates a lot of time consuming API calls to vmware.
-          if options.fetch('wait', true) then
-            # REVISIT: It would be awesome to call a block passed to this
-            # request to notify the application how far along in the process we
-            # are.  I'm thinking of updating a progress bar, etc...
-            new_vm = task.wait_for_completion
-          else
-            tries = 0
-            new_vm = begin
-              # Try and find the new VM (folder.find is quite efficient)
-              dest_folder.find(options['name'], RbVmomi::VIM::VirtualMachine) or raise Fog::Vsphere::Errors::NotFound
-            rescue Fog::Vsphere::Errors::NotFound
-              tries += 1
-              if tries <= 10 then
-                sleep 15
-                retry
+            # Waiting for the VM to complete allows us to get the VirtulMachine
+            # object of the new machine when it's done.  It is HIGHLY recommended
+            # to set 'wait' => true if your app wants to wait.  Otherwise, you're
+            # going to have to reload the server model over and over which
+            # generates a lot of time consuming API calls to vmware.
+            if options.fetch('wait', true) then
+              # REVISIT: It would be awesome to call a block passed to this
+              # request to notify the application how far along in the process we
+              # are.  I'm thinking of updating a progress bar, etc...
+              new_vm = task.wait_for_completion
+            else
+              tries = 0
+              new_vm = begin
+                # Try and find the new VM (folder.find is quite efficient)
+                dest_folder.find(options['name'], RbVmomi::VIM::VirtualMachine) or raise Fog::Vsphere::Errors::NotFound
+              rescue Fog::Vsphere::Errors::NotFound
+                tries += 1
+                if tries <= 10 then
+                  sleep 15
+                  retry
+                end
+                nil
```

Comment:
I think you could use Fog [`wait_for`](https://github.com/fog/fog-core/blob/master/lib/fog/core/wait_for.rb#L2) here.